src/reference/constellation/index.pug

```diff
@@ -0,0 +1,160 @@
+        li #[strong 4002] - Unknown packet type
+        li #[strong 4003] - Unknown method name call
+        li #[strong 4004] - Error parsing method arguments (not the right type or structure)
+        li #[strong 4005] - The user session has expired; if using a cookie, they should log in again, or get a bearer auth token if using an authorization header.
+        li #[strong 4100] - Unknown event in a livesubscribe call
+        li #[strong 4101] - You do not have access to subscribe to the livesubscribe event
+        li #[strong 4102] - You are already subscribed to the livesubscribe event (during livesubscribe)
+        li #[strong 4103] - You are not subscribe to the livesubscribe event (during liveunsubscribe)
+    h2#messages Messages
+    h3#messages_hello hello
+    p.
+        A hello event is sent down to the client when they first connect.
+    +highlightStr('json','{"type": "event", "event": "hello", "data": {"authenticated": true}}')
+    ul
+        li authenticated is true if the socket consumer authenticated as a user rather than a guest
+    h3#messages_livesubscribe livesubscribe
+    p.
+        A livesubscribe method is made available for users to listen to liveloading events.
+
+    +highlightStr('json','{"type": "method", "method": "livesubscribe", "params": {"events": ["user:1:update", "channel:1:followed"]}, "id": 42}')
+    ul
+        li  events is an array of events to subscribe to. Note that either all events are successfully subscribe to, or a failure occurs and no events are subscribed to. Either do or do not, there is no try.
+    p.
+        A livesubscribe reply looks like one of the following:
+    ul
+        li.
+            A successful reply:
+        +highlightStr('json','{"type": "reply", "result": null, "error": null, "id": 42}')
+        li.
+            A reply with an invalid event:
+        +highlightStr('json','{"type": "reply", "result": null, "error": {"code": 4100, "message": "Unknown event \'my silly event\' "}, "id": 42}')
+        li.
+            A reply for an event you do not have permission to see:
+        +highlightStr('json','{"type": "reply", "result": null, "error": {"code": 4101, "message": "Access denied on \'user:1:secrets\'"}, "id": 42}')
+        li.
+            A reply for an event you are already subscribed to:
+        +highlightStr('json','{"type": "reply", "result": null, "error": {"code": 4102, "message": "Attempt to duplicate subscription to \'user:1:update\'"}, "id": 42}')
+    h3#messages_liveunsubscribe liveunsubscribe
+    p.
+        A liveunsubscribe method can be used to stop listening to previously-subscribed-to events:
+    +highlightStr('json','{"type": "method", "method": "livesubscribe", "params": {"events": ["user:1:update", "channel:1:followed"]}, "id": 42}')
+    ul
+        li events is an array of events to unsubscribe from. Note that if you are not subscribed to one or more of the events, no error is returned.
```

Comment:
Yep that should be fixed. It will return an error, there's a test case for it...

![](https://peet.io/i/16-08-padjfj9ju293wcfyndbfmh6ju3i1tx.png)